teuthology/test/test_email_sleep_before_teardown.py

```diff
@@ -0,0 +1,81 @@
+from humanfriendly import format_timespan
+from mock import Mock, patch
+from pytest import mark
+from teuthology.config import config
+from teuthology.run_tasks import build_email_body as email_body
+from textwrap import dedent
+
+class TestSleepBeforeTeardownEmail(object):
+    def setup(self):
+        config.results_ui_server = "http://example.com/"
+        config.archive_server = "http://qa-proxy.ceph.com/teuthology/"
+
+    @mark.parametrize(
+        ['status', 'owner', 'suite_name', 'run_name', 'job_id', 'dura'],
+        [
+            [
+                'pass',
+                'noreply@host',
+                'dummy',
+                'run-name',
+                123,
+                3600,
+            ],
+            [
+                'fail',
+                'noname',
+                'yummy',
+                'next-run',
+                1000,
+                99999,
+            ],
+        ]
+    )
+    @patch("teuthology.run_tasks.time.time")
+    def test_sleep_before_teardown_email_body(self, m_time, status, owner,
+                                              suite_name, run_name, job_id, dura):
+        ctx = Mock()
+        archive_path='archive/path'
+        archive_dir='/archive/dir'
+        date_sec=3661
+        date_str='1970-01-01 01:01:01'
+        m_time.return_value=float(date_sec)
+        duration_sec=dura
+        duration_str=format_timespan(duration_sec)
+        ref_body=dedent("""
+            Teuthology job {run}/{job} has fallen asleep at {date} for {wait_time_sec} seconds
```

Comment:
instead of "for {wait_time_sec} seconds" I thought you would do

    for {duration_str}

to make it more humanfriendly?

Reply:
I can do that.